server/backend/db_backend.go

```diff
@@ -603,6 +606,47 @@ func (self *MongoBackend) updateContract(contract *models.Contract) bool {
 	return true
 }
 
+func (self *MongoBackend) getContracts(filter *models.ContractsFilter) []*models.Address {
+	var addresses []*models.Address
+	var sortQuery string
+	findQuery := bson.M{"contract": true}
+	if filter.TokenName != "" {
+		findQuery["token_name"] = bson.RegEx{filter.TokenName, "i"}
```

Comment:
Are we allowing users to submit arbitrary regex to the database? That seems exploitable.